node/primitives/src/lib.rs

```diff
@@ -258,3 +259,28 @@ impl std::convert::TryFrom<FromTableMisbehavior> for MisbehaviorReport {
 		}
 	}
 }
+
+/// The output of a collator.
+#[derive(Clone, Encode, Decode)]
+pub struct Collation {
+	/// Hash of `CandidateCommitments` as understood by the collator.
+	pub commitments_hash: Hash,
```

Comment:
I would prefer not to put the erasure-coding into the domain of Cumulus. That seems like a Polkadot-side task. The collation function can return upward messages, outgoing head data, etc. and then the collation generation subsystem can do erasure-coding, create the full candidate commitments, and yes, then just replace them with the hash as validators will just re-do the same computation.

Reply:
Ok: fd90cd40.